projects/hslayers/src/components/map/map.service.ts

```diff
@@ -791,23 +793,50 @@ export class HsMapService {
   repopulateLayers(visibilityOverrides: string[], app: string): void {
     const appConfig = this.hsConfig.get(app);
     if (appConfig.box_layers) {
+      let boxLayers: Layer[] = [];
       appConfig.box_layers.forEach((box) => {
-        for (const lyr of box.getLayers().getArray() as Layer<Source>[]) {
-          this.addLayer(
-            lyr,
-            app,
-            DuplicateHandling.IgnoreNew,
-            visibilityOverrides
-          );
-        }
+        boxLayers = boxLayers.concat(
+          (box.getLayers().getArray() as Layer<Source>[]).filter(
+            (layer) => layer
+          )
+        );
       });
+      this.addLayersFromAppConfig(boxLayers, visibilityOverrides, app);
     }
 
     if (appConfig.default_layers) {
-      const layers = appConfig.default_layers.filter((lyr) => lyr);
-      if (layers.length > 0) {
+      const defaultLayers: Layer[] = appConfig.default_layers.filter(
+        (lyr) => lyr
```

Comment:
Are you expecting a defaultLayers item to be falsy somehow? Other wise the filter would be not needed I think

Reply:
This was not my addition, so i am not sure, so I kept it for now.

Reply:
Oh, sorry. Just a safety measure 'what if'..I guess :D lets leave it than